modules/core/src/main/java/org/apache/synapse/message/store/impl/jms/JmsStore.java

```diff
@@ -639,6 +653,37 @@ private boolean initme() {
         return true;
     }
 
+    /**
+     * Use secure vault to secure password in JMS Message Store
+     * @param newParamValue
+     * @return
+     */
+    private String resolveSecureVaultExpressions(String newParamValue) {
+        Pattern vaultLookupPattern = Pattern.compile(secureVaultRegex);
+        Matcher lookupMatcher = vaultLookupPattern.matcher(newParamValue);
+        //setting value initially
+        if(lookupMatcher.find()) {
+            Value expression = null;
+            //getting the expression with out curly brackets
+            String expressionStr = lookupMatcher.group(1);
+            try {
+                expression = new Value(new SynapseXPath(expressionStr));
+            } catch (JaxenException e) {
+                log.error("Error while building the expression : " + expressionStr);
+            }
+            if (expression != null) {
```

Comment:
If the code block below line 674 is moved to line 671, this null check won't be needed. Also if JaxenException gets thrown while building this xpath, it looks like an unrecoverable situation, so we should throw a runtime exception (may be SynapseException, which is a RuntimeException) and then init() will fail in line 577, which is the expected behavior.